includes/User.php

```diff
@@ -2138,28 +2110,44 @@ public function setPassword( $str, $forceLogout = true ) {
 		return true;
 	}
 
+	private function heliosSetNewPassword( $password ) {
+		$heliosPasswordChange = null;
+
+		$heliosPasswordChange = self::heliosClient()->setPassword( $this->getId(), $password );
+
+		if ( empty( $heliosPasswordChange ) ) {
+			WikiaLogger::instance()->error( 'Helios password set communication failed', [
+				'userId' => $this->getId(),
+			] );
+			throw new PasswordError( wfMessage( 'externaldberror' )->text() );
+		}
+
+		if ( !empty( $heliosPasswordChange->errors ) ) {
+			WikiaLogger::instance()->error( 'Helios password set failed', [
+				'userId' => $this->getId(),
+				'err'    => $heliosPasswordChange,
+			] );
+			throw new PasswordError( wfMessage( $heliosPasswordChange->errors[0]->description )->text() );
+		}
+	}
+
+	private function heliosDeletePassword() {
+		self::heliosClient()->deletePassword( $this->getId() );
+	}
+
 	/**
 	 * Set the password and reset the random token unconditionally.
-	 *
-	 * @param $str String New password to set
 	 */
-	public function setInternalPassword( $str ) {
-		$this->load();
+	private function clearNewPasswordAndSetToken() {
 		$this->setToken();
 
-		if( $str === null ) {
-			// Save an invalid hash...
-			$this->mPassword = '';
-		} else {
-			$this->mPassword = self::crypt( $str );
-		}
 		$this->mNewpassword = '';
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Remove this hard-coded password. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS2068)